src-bindings/vscode/vscode.mli

```diff
@@ -1761,6 +1771,18 @@ module Workspace : sig
 
   val registerTextDocumentContentProvider :
     scheme:string -> provider:TextDocumentContentProvider.t -> Disposable.t
+
+  type workspaceFolderToAdd =
+    { name : string
+    ; uri : Uri.t
+    }
+
+  val updateWorkspaceFolders :
+       start:int
+    -> deleteCount:int or_undefined
```

Comment:
Is it correct? It can actually be `null`, `undefined` or `number`

Reply:
I think so? @mnxn would know better 

Ideally, `deleteCount` would be an optional argument, ie `... -> ?deleteCount:int -> ...`

Reply:
When going from JS to OCaml, `or_undefined` will turn the `number` into a `Some` `int` and `null` | `undefined` into `None`.

When going from OCaml to JS, `or_undefined` will turn a `Some` `int` into a `number` and `None` into `undefined`.

https://github.com/ocamllabs/vscode-ocaml-platform/blob/7870a8ecaaac8a81bd59764956880f54e6e095e6/src-bindings/interop/interop.ml#L7-L17

This means it's not possible to pass a `null` for this argument.

That being said, I don't think there's any reason to be able to pass both. The [VSCode documentation](https://code.visualstudio.com/api/references/vscode-api#workspace.updateWorkspaceFolders) doesn't say that it handles `null`s any differently than `undefined` here.

> Ideally, deleteCount would be an optional argument

@ulugbekna: This can't be done because [according to gen_js_api](https://github.com/LexiFi/gen_js_api/blob/master/TYPES.md#arrow-types):

> If no default value is provided and the argument is missing, the argument is dropped from the list of arguments passed to the JS call

If that argument was dropped, it would mess up the variadic arguments that follow it. Additionally, the VSCode documentation doesn't mark this parameter as optional like in the other places where we use OCaml optional arguments.

I think this should be left as-is.

Reply:
Oh and since there are no optional arguments in this function signature, you can remove the `unit` parameter.